### PR TITLE
Funnel devicelab tests through utils process methods

### PR DESCRIPTION
**Pull request by jmagman**
Devicelab tests [are failing with "lost connection"](https://github.com/flutter/flutter/issues/120808#issuecomment-1448934064), and it would be helpful to add screenshots when these commands fail (instead of adding a tricky-to-pipe `flutter run --screenshot` option, do it in the test harness).

Refactor devicelab tests to use the utils `startFlutter` etc. process methods so in my next PR I can add a hook for screenshots.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: dev/devicelab/lib/tasks/hot_mode_tests.dart [hunk 1]
```diff
@@ -22,7 +22,6 @@ const String kReplacementLine = 'fontSize: (orientation == Orientation.portrait)
 
 TaskFunction createHotModeTest({
   String? deviceIdOverride,
-  Map<String, String>? environment,
```
Comment (jmagman) [inline dev/devicelab/lib/tasks/hot_mode_tests.dart:25]: `environment` wasn't passed in anywhere, remove it.

File: dev/devicelab/lib/tasks/hot_mode_tests.dart [hunk 2]
```diff
@@ -63,71 +62,83 @@ TaskFunction createHotModeTest({
...
-            }
-            if (hotReloadCount == 0) {
-              // Update a file for 2 library invalidation.
-              final File appDartSource = file(path.join(
-                _editedFlutterGalleryDir.path, 'lib/gallery/app.dart',
-              ));
-              appDartSource.writeAsStringSync(
-                appDartSource.readAsStringSync().replaceFirst(
-                  "'Flutter Gallery'", "'Updated Flutter Gallery'",
+          smallReloadData = await captureReloadData(
```
Comment (jmagman) [inline dev/devicelab/lib/tasks/hot_mode_tests.dart:65]: This part is mostly whitespace changes.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**